.github/workflows/publish-packages.yml

```diff
@@ -89,7 +89,7 @@ jobs:
 
       - name: Build all packages
         run: |
-          npm run build -- --filter='./packages/*'
+          NODE_ENV="production" npm run build -- --filter='./packages/*'
```

Comment:
Not really sure how I feel about relying on NODE_ENV for this.

Would it be an option to simply always let the declaration maps be generated (also in production), but simply not include them in the NPM bundle by excluding their file extension?

```
  "files": [
    "dist/**",
    "!**.d.ts.map",  👈
    "README.md"
  ],
```

Reply:
This exclude pattern works, I just tested it locally 👍

Reply:
The biggest reason is that I'm not 100% confident that the different `.d.ts` outputs don't lead to differences, but if they don't, I'll find a way to duplicate them as `.d.cts` and I agree having a single path would be easier and more predictable

Reply:
I tried to simplify things with the "single output" option but I think I've reached a dead-end: generating the `.d.ts` files with `tsc` separately from the code leads to different file structures and publint links that to an "Internal resolution error" with `node16` resolution (see image in PR description) 😢 Fixing this seems like a rabbit hole I don't want to go into atm but there's still the option of two outputs:
- local: break `node16` resolution but gain declaration maps
- production: same output as before this PR
And we could use something else than `process.env.NODE_ENV` to switch between local and production